0000-template.md

```diff
@@ -1,51 +1,48 @@
 
-# Motivation
+# Motivation and background
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Why are we doing this? What use cases does it support? What is the expected outcome? Why are existing solutions not good enough? If possible, include data to support your claims. Include any background context useful for understanding the RFC.
 
-# Guide-level explanation
-[guide-level-explanation]: #guide-level-explanation
+# Detailed explanation
+[detailed-explanation]: #detailed-explanation
 
-Explain the proposal as if it was already included in the language and you were teaching it to another Rust programmer. That generally means:
+Explain the proposal in detail. Specify the proposal as it would be experienced by a user (usually a Rust developer) and so that it can be understood by an implementer. That generally means:
 
 - Introducing new named concepts.
-- Explaining the feature largely in terms of examples.
+- Explaining the feature with examples.
 - Explaining how Rust programmers should *think* about the feature, and how it should impact the way they use Rust. It should explain the impact as concretely as possible.
 - If applicable, provide sample error messages, deprecation warnings, or migration guidance.
 - If applicable, describe the differences between teaching this to existing Rust programmers and new Rust programmers.
-
-For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
-
-# Reference-level explanation
-[reference-level-explanation]: #reference-level-explanation
-
-This is the technical portion of the RFC. Explain the design in sufficient detail that:
-
-- Its interaction with other features is clear.
+- If applicable, how the feature will be taught, documented, and tested.
 - It is reasonably clear how the feature would be implemented.
+- Its interaction with other features is clear.
 - Corner cases are dissected by example.
 
-The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
-
-# Drawbacks
-[drawbacks]: #drawbacks
+For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
 
-Why should we *not* do this?
+# Rationale
+[rationale]: #rationale
 
-# Rationale and alternatives
-[rationale-and-alternatives]: #rationale-and-alternatives
+Discuss the *why* of your proposal. You might want to include the following:
```

Comment:
I think saying "discuss the why of your proposal" might be a little too vague or confusing with the "Motivation" section, as that also discuses the *why* of the RFC. Perhaps that can be reworded to more specifically state that this section is for why *this* proposal and not some other?

Some time ago I wrote my own new outline, and it was largely similar except I merged "Alternatives and Prior Art" (and removed "Rationale"). My thinking is that "why choose this over something else?" applies to all the various different questions listed in these sections, so it seemed like a single section would be sufficient.

Reply:
I had a go at rewording. My feeling is that discussing rationale is general enough to be a useful section and important to encourage discussing. Prior art feels a bit different to me since it is kind of 'supporting evidence' from other communities or history, rather than about reflecting on the proposal.